plugin/forward/forward.go

```diff
@@ -109,7 +109,7 @@ func (f *Forward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 		opts := f.opts
 		for {
 			ret, err = proxy.Connect(ctx, state, opts)
-			if err == nil {
+			if err == nil && ret != nil && !ret.Truncated {
```

Comment:
I would suggest to remove this `if` block at all. This would also fix the problem.